# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/OriginAttributes.cpp
# Commit: 336214ad7891
# Full Hash: 336214ad789156999085eb7607fe6a6cbf7bbfe9
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-10 21:44:02
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 453324c772ae -r 336214ad7891 caps/OriginAttributes.cpp
--- a/caps/OriginAttributes.cpp	Fri Jan 10 08:24:29 2025 +0000
+++ b/caps/OriginAttributes.cpp	Fri Jan 10 08:24:29 2025 +0000
@@ -61,7 +61,7 @@
                                         bool aForeignByAncestorContext,
                                         bool aIsFirstPartyEnabled, bool aForced,
                                         bool aUseSite,
-                                        nsString OriginAttributes::*aTarget,
+                                        nsString OriginAttributes::* aTarget,
                                         OriginAttributes& aOriginAttributes) {
   nsresult rv;
 
@@ -514,4 +514,23 @@
   return fieldIndex > 1;
 }
 
+/* static */
+bool OriginAttributes::ExtractSiteFromPartitionKey(
+    const nsAString& aPartitionKey, nsAString& aOutSite) {
+  nsAutoString scheme, host;
+  int32_t port;
+  bool unused;
+  if (!ParsePartitionKey(aPartitionKey, scheme, host, port, unused)) {
+    return false;
+  }
+
+  if (port == -1) {
+    aOutSite.Assign(scheme + u"://"_ns + host);
+  } else {
+    aOutSite.Assign(scheme + u"://"_ns + host + u":"_ns);
+    aOutSite.AppendInt(port);
+  }
+  return true;
+}
+
 }  // namespace mozilla